src/routes/balances/balances.service.ts

```diff
@@ -37,6 +43,10 @@ export class BalancesService {
     trusted: boolean;
     excludeSpam: boolean;
   }): Promise<Balances> {
+    if (this.pricesProviderChainIds.includes(args.chainId)) {
+      return this._getFromSimpleBalances(args);
+    }
```

Comment:
What is the main reason for having a decision point on the `routes` layer? I don't think this addressed directly the concern regarding how the balances are abstracted 🤔 

If the domain entities (we still get a `Balance` object) why did the logic on the routes had to change?

The lines below this statement should still be valid right? More specifically:
- Mapping the balances to the Route entities
- Getting the total fiat from the mapped balances
- Sorting the balances

Reply:
I tried to address this in my previous comment (https://github.com/safe-global/safe-client-gateway/pull/732#discussion_r1350906338): the decision point still in the routes layer because otherwise we would need to introduce some changes in the previously existing code:
- `IExchangeRepository` injection would be moved to `BalancesRepository`, so we can retrieve/apply fiat conversion rates there.
- (derived from above point) Some logic that is now part of the mapping function (`usdToFiatRate` multiplication, for example), should be moved to the repository too, because it depends on whether we are using the Exchange API or Coingecko API.

So I’d say it’s not possible to keep all the lines you mentioned. I opted for the strategy of having separated functions and an if condition for routing, because the feature flag should be fully applied soon and the Exchange-related function would be removed easily. So perhaps this strategy results in redundant code for a while, but would also be easy to clean later, after the FF is fully applied.

Having said that, I’m open to move all the logic to `BalancesRepository` as you suggested, even if it implies modifying existing code. I leave this up to you, what do you think? Would you push all the logic into the repository in this PR? :slightly_smiling_face: (

Reply:
After a quick sync, I've applied your suggestion in [35029ea](https://github.com/safe-global/safe-client-gateway/pull/732/commits/35029ea40c3cf5243b32eeeec1f1b296b9db2e6a), thanks!